### ISSUE TITLE
pdf x-change editor

### ISSUE THREAD

**Issue by eiselesr** [jlegewie/zotfile#239]
When extracting annotation from pdfs in pdf x-change editor the text from highlights, sticky notes and pop-up notes are captured but if Text box annotations are used these are not picked up by zotfile. 

I believe it is possible to capture them since docear is able to, but I am not sure how to go about capturing them.  Do you have any suggestions for adding that functionality?

**Comment by jlegewie** [comment]
Yes, that should be possible but I never looked into it. I also won't do it. You can take a look though but it does require some knowledge. `chrome/content/zotfile/pdfextract/pdfjs/src/getPDFAnnotations.js` filters out the supported annotations (see variable `SUPPORTED_ANNOTS`, which is set to `["Text", "Highlight", "Underline"]`). But it's going to be more work than just changing that variable.

**Comment by eiselesr** [comment]
Thanks for your quick response.
Could you point me at what else might need changing? js is pretty new to
me.

I appreciate any guidance you can provide.

-Scott Eisele

**Comment by eiselesr** [comment]
I believe I'd have to add to the annotation.js script for freetext
annotations.

// TODO(mack): Implement FreeText annotations
    if (subtype === 'Link') {
      return LinkAnnotation;
    } else if (subtype === 'Text') {
      return TextAnnotation;
    } else if (subtype === 'Widget') {
      if (!fieldType) {
        return;
      }

Is there anything else? Or would this be a better question for the pdfjs
guys?

**Comment by jlegewie** [comment]
With that change, you are actually changing pdf.js core. Not necessarily a good idea. Zotfile does that but it should be reduced to a minimum. You should try get the annotations (that is what `page.getAnnotations()` in `getPDFAnnotations.js` does, find an object with the annotation type you are interested in and see whether it already contains the text you need. You can either figure that out by actually looking at the javascript object or by looking at the pdf.js documentation (or if that doesn't exist code).
